pages/api/paybutton/download/transactions/[paybuttonId].ts

```diff
@@ -0,0 +1,136 @@
+import Cors from 'cors'
+import moment from 'moment'
+import {
+  CAD_QUOTE_ID,
+  DEFAULT_QUOTE_SLUG,
+  PRICE_API_DATE_FORMAT,
+  USD_QUOTE_ID,
+  SUPPORTED_QUOTES,
+  RESPONSE_MESSAGES,
+  DEFAULT_PAYBUTTON_TRANSACTIONS_FILE_DELIMITER,
+  PAYBUTTON_TRANSACTIONS_FILE_HEADERS
+} from 'constants/index'
+import { TransactionWithAddressAndPrices, fetchTransactionsByPaybuttonId } from 'services/transactionService'
+import { PaybuttonWithAddresses, fetchPaybuttonById } from 'services/paybuttonService'
+import { streamToCSV } from 'utils/files'
+import { setSession } from 'utils/setSession'
+import { NextApiResponse } from 'next'
+import { TransactionFileData, FormattedTransactionFileData } from 'types'
+type SupportedQuotesType = typeof SUPPORTED_QUOTES[number]
+
+const cors = Cors({ methods: ['GET', 'HEAD'] })
+
+const getPrice = (transaction: TransactionWithAddressAndPrices, currency?: SupportedQuotesType): number => {
+  let stringCurrency: string = currency ?? ''
+  const isCurrencyEmptyOrUndefined = currency === '' || currency === undefined
+
+  if (isCurrencyEmptyOrUndefined) {
+    stringCurrency = DEFAULT_QUOTE_SLUG
+  }
+  const { prices, amount } = transaction
+  const result: { [key in SupportedQuotesType]: number } = {}
+
+  for (const p of prices) {
+    if (p.price.quoteId === USD_QUOTE_ID) {
+      result.usd = p.price.value.times(amount).toNumber()
+    }
+    if (p.price.quoteId === CAD_QUOTE_ID) {
+      result.cad = p.price.value.times(amount).toNumber()
+    }
+  }
+  return (!isCurrencyEmptyOrUndefined ? result[stringCurrency] : result.usd)
+}
+
+const getPaybuttonTransactionsFileData = (transaction: TransactionWithAddressAndPrices, paybutton: PaybuttonWithAddresses): TransactionFileData => {
+  const { amount, createdAt, id } = transaction
+  const value = getPrice(transaction)
```

Comment:
`value = getPrice` reinforces the point I made in the comment above about the function `getPrice` renaming